### PR TITLE
Implement finalise track road ordering

### PR DESCRIPTION
**Pull request by duncanspumpkin**
I know in the meeting we said paint wasn't the priority but i was wondering how something worked and ended up implementing this.

### CODE REVIEW

File: src/OpenLoco/src/Paint/Paint.cpp [hunk 1]
```diff
@@ -1172,4 +1172,168 @@ namespace OpenLoco::Paint
...
+        routeEntry->bounds.x = minMaxBounds.x;
+        routeEntry->bounds.y = minMaxBounds.y;
+        routeEntry->bounds.xEnd = minMaxBounds.xEnd;
+        routeEntry->bounds.yEnd = minMaxBounds.yEnd;
+
+        addPSToQuadrant(*routeEntry);
+    }
+
+    // 0x0045CC1B
+    void PaintSession::finaliseTrackRoadAdditionsOrdering()
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Paint/Paint.cpp:1259]: these two are very similar should just merge them into one common function.

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Paint/Paint.cpp:1259]: Looks like only the destination addresses differ (`_E400D0` vs `_E400E4`). Is this something you'd like to parametrise out in this PR?

File: src/OpenLoco/src/Paint/Paint.h [hunk 2]
```diff
@@ -345,8 +347,8 @@ namespace OpenLoco::Paint
         inline static Interop::loco_global<PaintStruct* [kMaxPaintQuadrants], 0x00E3F0C0> _quadrants;
         inline static Interop::loco_global<uint32_t, 0x00E400C0> _quadrantBackIndex;
         inline static Interop::loco_global<uint32_t, 0x00E400C4> _quadrantFrontIndex;
-        inline static Interop::loco_global<PaintEntry* [2], 0x00E400E4> _E400E4;
-        inline static Interop::loco_global<PaintEntry* [5], 0x00E400D0> _E400D0;
+        inline static Interop::loco_global<PaintStruct* [2], 0x00E400E4> _E400E4;
+        inline static Interop::loco_global<PaintStruct* [5], 0x00E400D0> _E400D0;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Paint/Paint.h:351]: I believe these can now be named, right?

**Review by AaronVanGeffen — COMMENTED** [review]
There seems to be an issue drawing certain bridges. See screenshot below.

Save file: [Great Britain & Ireland 1936d.SV5.zip](https://github.com/OpenLoco/OpenLoco/files/13628250/Great.Britain.Ireland.1936d.SV5.zip)

Screenshot 2023-12-10 at 16 18 03

File: src/OpenLoco/src/Paint/Paint.cpp [hunk 3]
```diff
@@ -1172,4 +1173,116 @@ namespace OpenLoco::Paint
...
+        return routeEntry;
+    }
+
+    // 0x0045CABF, 0x0045CAF4, 0x0045CB29, 0x0045CB5A, 0x0045CC73, 0x0045CCA8, 0x0045CCDD, 0x0045CD0E
+    static PaintStructBoundBox getMinMaxXYBounding(PaintStruct& routeEntry, uint8_t rotation)
+    {
+        auto minMaxBounds = routeEntry.bounds;
+        switch (rotation)
+        {
+            case 0:
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Paint/Paint.cpp:1218]: I wonder if we should introduce constants for these?

**Review by AaronVanGeffen — APPROVED** [review]
Looks like it's working properly now. :+1:

**Pull request merged**